l1-contracts/contracts/bridgehub/MessageRoot.sol

```diff
@@ -53,6 +56,11 @@ contract MessageRoot is IMessageRoot, Initializable {
     /// @dev The incremental merkle tree storing the chain message roots.
     mapping(uint256 chainId => DynamicIncrementalMerkle.Bytes32PushTree tree) internal chainTree;
 
+    /// @notice The mapping from block number to the global message root.
+    // kl todo this approach does not work, each block might have multiple txs that bump the historical root.
```

Comment:
do we need to fix it in this PR?

Reply:
I think yes. The best solution is to link MessageRoots with the previous one, it solves other problems as well

Reply:
discusses, we will just overwrite and load in the latest value in EthWatch